test/e2e/scrapeconfig_test.go

```diff
@@ -2119,3 +2122,148 @@ var LightSailSDTestCases = []scrapeCRDTestCase{
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid Service type",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "InvalidService",
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Empty Endpoint",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+					Endpoint:          ptr.To(""),
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid with RefreshInterval for supported version",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+			OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+				{
+					ApplicationKey:    "valid-app-key",
+					ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+					ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+					Service:           "VPS",
+					RefreshInterval:   ptr.To(monitoringv1.Duration("30s")),
+				},
+			},
+		},
+		expectedError: false,
+	},
+	// {
+	// 	// Following should technically show error when the refreshInterval has invalid value for the supported version
+	// 	name: "Invalid RefreshInterval for unsupported version",
+	// 	scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+	// 		OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+	// 			{
+	// 				ApplicationKey:    "valid-app-key",
+	// 				ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+	// 				ConsumerKey:       v1.SecretKeySelector{Key: "valid-consumer-key"},
+	// 				Service:           "VPS",
+	// 				RefreshInterval:   ptr.To(monitoringv1.Duration("30s")),
+	// 			},
+	// 		},
+	// 	},
+	// 	expectedError: true,
+	// },
+	// {
+	// 	name: "Missing ConsumerKey",
+	// 	scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+	// 		OVHCloudSDConfigs: []monitoringv1alpha1.OVHCloudSDConfig{
+	// 			{
+	// 				ApplicationKey:    "valid-app-key",
+	// 				ApplicationSecret: v1.SecretKeySelector{Key: "valid-secret-key"},
+	// 				Service:           "VPS",
+	// 			},
+	// 		},
+	// 	},
+	// 	expectedError: true,
+	// },
+	// {
+	// 	name: "Invalid ConsumerKey with empty value",
```

Comment:
This test is not testing for expected behavior. Right now the test fails because, `Key` has not been defined as a required field.

@simonpasquier, I think we can keep `Key` an optional field. I reckon we can make the `LocalObjectReference` a required field though. WDYT?
cc: @slashpai

Reply:
And by the looks of it, that is not up to us. We are consuming K8S's API. See:  https://github.com/kubernetes/api/blob/v0.32.0/core/v1/types.go#L2430-L2438.